readthedocs/core/views/hooks.py

```diff
@@ -88,3 +90,21 @@ def sync_versions(project):
     except Exception:
         log.exception('Unknown sync versions exception')
     return None
+
+
+def get_or_create_external_version(project, identifier, verbose_name):
+    external_version = project.versions(manager=EXTERNAL).filter(verbose_name=verbose_name).first()
+    if external_version:
+        if external_version.identifier != identifier:
```

Comment:
This will be updated when there is a new commit in the PR

Reply:
This is probably better in a code comment, not a PR comment that will be lost :)